lib/shoulda/matchers/active_record/have_attached_matcher.rb

```diff
@@ -0,0 +1,133 @@
+module Shoulda
+  module Matchers
+    module ActiveRecord
+      def have_one_attached(name)
+        HaveAttachedMatcher.new(:one, name)
+      end
+
+      def have_many_attached(name)
+        HaveAttachedMatcher.new(:many, name)
+      end
+
+      # @private
+      class HaveAttachedMatcher
+        attr_reader :name
+
+        def initialize(macro, name)
+          @macro = macro
+          @name = name
+        end
+
+         def description
+          "has_#{macro}_attached #{name}"
+        end
+
+        def failure_message
+          "Expected #{expectation} (#{@failure})"
```

Comment:
This is good, but in a future update I'm going to be updating the messaging for the association matchers so they align with other matchers. The work I did in 47f27f01810e14bbf43705f67abdcfa7b2e9b08c I think serves as a good example. What do you think about something like this:

* If the reader attribute does not exist:
  > Expected Gallery to have many attached :photos, but this could not be proved.
  >   Gallery does not have a :photos method.
* If the writer attribute does not exist:
  > Expected Gallery to have many attached :photos, but this could not be proved.
  >   Gallery does not have a :photos= method.
* If either attachments or blobs association does not exist, you can re-use the failure message from the association matcher.
* If the eager loading scope doesn't exist:
  > Expected Gallery to have many attached :photos, but this could not be proved.
  >   Gallery does not have a .with_attached_photos method.